src/pages/Florist/index.tsx

```diff
@@ -0,0 +1,237 @@
+import styles from "./Florist.module.css";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    container: {
+      width: "90%",
+      marginLeft: "2.5%",
+      padding: "5%",
+    },
+    header: {
+      margin: "2%",
+      marginLeft: "5%",
+      width: "100%",
+    },
+    flowerList: {
+      width: "100%",
+    },
+    card: {
+      margin: "1%",
+      height: "30%",
+      width: "30%",
+      display: "flex",
+      flexDirection: "column",
+      alignItems: "center",
+    },
+    cardBought: {
+      margin: "1%",
+      height: "30%",
+      width: "30%",
+      display: "flex",
+      flexDirection: "column",
+      alignItems: "center",
+      backgroundColor: "#6ac69780",
+    },
+    buyButton: {
+      marginBottom: "0.125rem",
+      width: "100%",
+    },
+    errorMsg: {
+      backgroundColor: theme.palette.error.light,
+    },
+  })
+);
+
+export const Florist = () => {
+  const classes = useStyles();
+  const { userData, setUserData } = useUserState();
+  const [lastBought, setLastBought] = useState<string | null>(null);
+  const [flowerPrice, setFlowerPrice] = useState(0);
+  const [buyFlowerError, setBuyFlowerError] = useState(false);
+  const [errMsgVis, setErrMsgVis] = useState(false);
+
+  const [buyFlowerAPIState, buyFlowerReq] = useApi(buyFlower);
+
+  const userId = useMemo(
+    () => (userData.isLoggedIn && userData.id) || "",
+    [userData]
+  );
+  const boughtFlowers = useMemo(
+    () => (userData.isLoggedIn && userData.flowerCollections) || [],
+    [userData]
+  );
+
+  const buyFlowerHandler = useCallback(
+    async (flowerId: string, flowerPrice: number) => {
+      if (userId) {
+        setLastBought(flowerId);
+        setFlowerPrice(flowerPrice);
+        await buyFlowerReq({
+          fireBaseUserId: userId,
+          flowerId,
+          price: flowerPrice,
+        });
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [userId]
+  );
+
+  useEffect(() => {
+    if (buyFlowerAPIState.status === "succeeded" && lastBought) {
+      setBuyFlowerError(false);
+
+      setUserData((data) => {
+        const newFlowerCollection =
+          (data.isLoggedIn && data.flowerCollections) || [];
+
+        const currentBalance = (data.isLoggedIn && data.balance) || 0;
+        return {
+          ...data,
+          balance: currentBalance - flowerPrice,
+          flowerCollections: [...newFlowerCollection, ...[lastBought]],
+        };
+      });
+      // MEMO: Conall not sure about this logic so if needed uncomment it else remove it
+      // setBought((prev) => {
+      //   if (!bought.includes(lastBought)) {
+      //     return [...prev, lastBought];
+      //   } else {
+      //     return prev;
+      //   }
+      // });
+    }
+
+    if (buyFlowerAPIState.status === "failed") {
+      setBuyFlowerError(true);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [buyFlowerAPIState.status, lastBought, flowerPrice]);
```

Comment:
The issue was using global state and adding it to the deps, so changed to use the current state directly from setUserData